app/controllers/mixins/ems_common_angular.rb

```diff
@@ -93,7 +93,12 @@ def create_ems_button_validate
       result, details = if %w(ems_cloud ems_infra).include?(params[:controller]) && session[:selected_roles].try(:include?, 'user_interface')
                           realtime_raw_connect(ems_type)
                         elsif %w(ems_cloud ems_infra).include?(params[:controller])
-                          ems_type.validate_credentials_task(get_task_args(ems_type), session[:userid], params[:zone])
+                          if params[:cred_type] == 'amqp'
```

Comment:
This is wrong.

That condition needs to live in `ems_type.validate_credentials_task` for your provider class.

We can not introduce uses of provider-specific services in the UI.

Reply:
@himdel, right, meanwhile here the issue is the task for `raw_connect?` is not the one needed (it's an AMQP event service and not a cloud/infra provider). 

Therefore we have to make such decision before and since there is no layer to handle AMQP tasks maybe in the meantime we fix this bug and add just a little bit more provider stuff to the existing ones (which don't make it right, I agree).

Please don't kill the messenger ;)

Reply:
@Ladas, @aufi, your thoughts?

Reply:
I'm with @himdel here...at a minimum, the next 2 lines must live in the provider and the provider could expose an amqp_available? method.  Even so, the way it's written now is synchronous, and it has to be async...the UI may not have connectivity to the provider, so extending `validate_credentials_task` would be the better place to start.

Reply:
@Fryguy, makes sense, I'll look into that, thanks!